proto_proxy.c

```diff
@@ -256,33 +227,59 @@ void proxy_submit_cb(io_queue_t *q) {
         } else {
             q->count++;
         }
+        be = p->backend;
+
+        if (be->use_io_thread) {
```

Comment:
Think this is the section I was agonizing over the most. It's honestly good enough for now at least, I just wish I could reuse more of the `proxy_event_handler` code.

This is more efficient than reusing the event handler code since that loops a second time to split queues onto backend objects.

Once I split the backend IO queues into to-write and to-read it's a little tempting to make the to-write queue locked, so they don't have to be iterated again post-submission.